File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/ValidUpdateStackRequest.java
package com.sequenceiq.cloudbreak.controller.validation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import javax.validation.Constraint;
import javax.validation.Payload;

@Target({ ElementType.TYPE })
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = UpdateStackRequestValidator.class)
public @interface ValidUpdateStackRequest {

    String message() default "Update stack request is not valid.";

    Class<?>[] groups() default { };

    Class<? extends Payload>[] payload() default { };

}

File: core-api/src/main/java/com/sequenceiq/cloudbreak/validation/ValidUpdateStackRequest.kt
package com.sequenceiq.cloudbreak.validation

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = UpdateStackRequestValidator::class)
annotation class ValidUpdateStackRequest(val message:

                                         String = "Update stack request is not valid.", val groups: Array<KClass<*>> = arrayOf(), val payload: Array<KClass<out Payload>> = arrayOf())